server/neptune/workflows/internal/deploy/revision/queue/worker.go

```diff
@@ -61,15 +76,66 @@ func (w *Worker) Work(ctx workflow.Context) {
 
 		msg := w.Queue.Pop()
 
-		ctx := workflow.WithValue(ctx, context.SHAKey, msg.Revision)
-		ctx = workflow.WithValue(ctx, context.DeploymentIDKey, msg.ID)
+		ctx := workflow.WithValue(ctx, internalContext.SHAKey, msg.Revision)
+		ctx = workflow.WithValue(ctx, internalContext.DeploymentIDKey, msg.ID)
 
-		err = w.TerraformWorkflowRunner.Run(ctx, msg)
+		// This should only happen on startup
+		// If we fail to fetch the latest deployment, log and exit the workflow
+		if w.LatestDeployment == nil {
+			w.LatestDeployment, err = w.fetchLatestDeployment(ctx, msg)
+			if err != nil {
+				return
```

Comment:
ok sorry about this, i misinterpreted the logging line you had before and thought it was something else lol, feel free to add it back like the other error statements 😅 . Sorry there's a lot going on.

Reply:
no worries!